src/services/blocks/BlocksService.ts

```diff
@@ -700,8 +700,15 @@ export class BlocksService extends AbstractService {
 					newArgs[paramName] = this.parseArrayGenericCalls(argument, registry);
 				} else if (argument instanceof GenericCall) {
 					newArgs[paramName] = this.parseGenericCall(argument, registry);
-				} else if (paramName === 'call' && argument?.toRawType() === 'Bytes') {
-					// multiSig.asMulti.args.call is an OpaqueCall (Vec<u8>) that we
+				} else if (
+					argument &&
+					paramName === 'call' &&
+					['Bytes', 'WrapperKeepOpaque<Call>', 'WrapperOpaque<Call>'].includes(
+						argument?.toRawType()
+					)
+				) {
+					// multiSig.asMulti.args.call is either an OpaqueCall (Vec<u8>),
+					// WrapperKeepOpaque<Call>, or WrapperOpaque<Call> that we
```

Comment:
It's a bit hard to map the code in the `try…catch` block to each of the three cases. What do you think about something like this:

```ts
			} else if (
				argument &&
				paramName === 'call'
			) {
				switch (argument?.toRawType()) {
					case 'Bytes':
					case 'WrapperKeepOpaque<Call>':	
						newArgs[paramName] = argument;
					case 'WrapperOpaque<Call>':
						const call = registry.createType('Call', argument.toHex());
						newArgs[paramName] = this.parseGenericCall(call, registry);
	
					default:
						// log or otherwise report error?;
				}
```

(Not sure I got the cases right, which is illustrating the point I guess)

Reply:
I like that if that works :)

Reply:
So actually each of the 3 cases are going to map out to the same output i.e.
```
					try {
						const call = registry.createType('Call', argument.toHex());
						newArgs[paramName] = this.parseGenericCall(call, registry);
					} catch {
						newArgs[paramName] = argument;
					}
```

therefore the above wont work. I filed an issue [here](https://github.com/paritytech/substrate-api-sidecar/issues/845) regarding the `try catch` in order to potentially optimize it. For this PR i think it is out of scope as we are just extending the conditional to include `WrapperKeepOpaque<Call>`, and `WrapperOpaque`.